lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -244,8 +226,119 @@ def _run(self, params, concierge_params=None):
         self.logger.debug(
-            f"Attempting to update job to queued  {job_id} {condor_job_id} {submission_info}"
+            f"Attempting to update job to queued  {job_id} {condor_job_id}"
+        )
+
+    # def _cache_catalog_resources(self, param_set):
+    #     # Get a list of all of the methods
+    #     # Get their clientgroups and resources from the catalog
+    #     # Cache method resources and versions
+    #     mr_cache = self.catalog_utils.get_mass_resources()
+    #     mv_cache = dict()
+    #     for param in param_set:
+    #         method = param['method']
+    #         service_ver = param['service_ver']
+    #         # Get resources for method
+    #         if method not in mr_cache:
+    #             self.sdkmr.get_condor().extract_resources()
+    #             mr_cache[method] = self.catalog_utils.get_normalized_resources(method=method)
+    #         # Get Method Version via Git Commit
+    #         if method not in mv_cache and service_ver not in mv_cache[method]:
+    #             mv_cache[method][service_ver]= self._get_module_git_commit(method=method,
+    #                                                                        service_ver=service_ver)
+    #
+    #     return CatalogUtils
+
+    # TODO delete this after checking it in
+    # def _get_cached_catalog_resources(
+    #     self, job_param_set: List[dict]
+    # ) -> JobResourceMappings:
+    #     # Check workspace objects for each of the jobs
+    #     workspace_check_time = time.time()
+    #     [
+    #         self._check_ws_objects(param.get("source_ws_objects"))
+    #         for param in job_param_set
+    #     ]
+    #     self.logger.debug(
+    #         f"Took {time.time() - workspace_check_time} to check all workspace objects"
+    #     )
+    #     #Get a mapping of condor resources
+    #     catalog_check_time = time.time()
+    #     condor_resources_mapping = self.catalog_utils.get_mass_resources(
+    #         job_param_set=job_param_set, condor=self.condor
+    #     )
+    #     self.logger.debug(
+    #         f"Took {time.time() - catalog_check_time} to get all condor resources"
+    #     )
+    #     # Get a mapping of method git versions. If one is not valid, throw the catalog error #TODO Test this
+    #     catalog_check_time_git = time.time()
+    #
+    #
+    #     return JobResourceMappings(
+    #         condor_resources=condor_resources_mapping,
+    #         method_service_versions=service_versions_mapping,
+    #     )
+
+    def _evaluate_job_params_set(self, job_param_set, parent_job):
+        # Fail early before job submission if possible
+        # Make sure clientgroups and condor resources are set correctly for each job,
+        # or raise an exception if something goes wrong
+        for job_params in job_param_set:
+            self.catalog_utils.get_git_commit_version(job_params=job_params)
+            self.catalog_utils.get_condor_resources(
```

Comment:
These two lines are relying on the called methods to throw an exception if there's something wrong with the job params, right? It'd be good to comment to that effect if so

Reply:
Updated comment for `_evaluate_job_params_set` to add `if job params are incorrect` (or are you saying that each time we call a function we have to say that we rely that the function below to throw an exception if something is going wrong?)

Reply:
I'm saying that since based on their names a user would assume the functions just fetch data and said data is not used at all a coder might assume they're cruft and delete them, so it might be worthwhile to put in a comment explaining why they're being called.